tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -409,4 +410,27 @@ var _ = Describe("odo devfile create command tests", func() {
 			Expect(stdout).To(ContainSubstring("Please use `odo push` command to create the component with source deployed"))
 		})
 	})
+
+	When("there is already a devfile in the directory", func() {
+		BeforeEach(func() {
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+		})
+		It("should fail with appropriate error when --starter is given to odo create", func() {
+			stderr := helper.Cmd("odo", "create", "--starter", "nodejs-starter", "--context", commonVar.Context, "--project", commonVar.Project).ShouldFail().Err()
+			Expect(stderr).To(ContainSubstring("already has a devfile so you cannot provide a starter"))
+		})
+	})
+
+	When("a devfile is provided which has a starter that has its own devfile", func() {
+		BeforeEach(func() {
+			examplesPath := helper.GetExamplePath()
+			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, "--context", commonVar.Context, "--starter", "nodejs-starter", "--devfile", filepath.Join(examplesPath, "source", "devfiles", "nodejs", "devfile-with-starter-with-devfile.yaml")).ShouldPass()
```

Comment:
@mohammedzee1000 
When downloading the starter project, the target directory will be emptied before to download the starter files. Because the configuration files are placed in the context directory, these config files will be deleted and this is what makes your tests fail.

To fix the problem, you can create a second temporary directory for each test where to place config files, different of the context directory. You can see what I have done in the PR https://github.com/redhat-developer/odo/pull/5348/files (in the file tests/helper/helper_generic.go)

Reply:
Ah finally, thanks @feloy

Reply:
It would be simpler to use the changes I have made in the PR #5348 that is being merged (or wait for the PR to be merged and rebase your branch)